lib/phoenix_ecto/check_repo_status.ex

```diff
@@ -10,13 +10,14 @@ defmodule Phoenix.Ecto.CheckRepoStatus do
     * `:otp_app` - name of the application which the repos are fetched from
     * `:migration_paths` - a function that accepts a repo and returns a migration directory, or a list of migration directories, that is used to check for pending migrations
     * `:migration_lock` - the locking strategy used by the Ecto Adapter when checking for pending migrations. Set to `false` to disable migration locks.
+    * `:prefix` - the prefix used to check for pending migrations.
```

Comment:
Instead of `prefix`, what if we allow `migration_paths` to be `{path, opts}` pairs? This way you can specify opts per path?

Reply:
Yeah, that would be more flexible & wouldn't require multiple plug configurations.

```
defmodule MyMod do
  def migrations_paths(repo) do
    default_migrations_path = Migrator.migrations_path(repo)

    tenant_migrations_path =
      Path.join(default_migrations_path, "../tenant_migrations") |> Path.expand()

    [default_migrations_path, {tenant_migrations_path, prefix: "tenant_acme"}]
  end
end

# endpoint

    plug Phoenix.Ecto.CheckRepoStatus,
      otp_app: :myapp,
      migration_paths: &MyMod.migrations_paths/1
```

Could also fetch the tenant schema names from the db rather than use a fixed value. Will update tomorrow 😄

Reply:
I've updated the PR to allow tuples in `migration_paths`. There's a slight change in behaviour in [5648d73](https://github.com/phoenixframework/phoenix_ecto/pull/181/commits/5648d730ba722e3af111be0feb37a568fb182aae), where given `fn -> ["a", "b"] end` it'll run the migrator once for `a` and again for `b`.

I'm not sure how well the option name `migration_paths` fits with this new functionality. Alternative names could be `migrate_runs` or a `migrate: fn -> ["a", {"b", prefix: "tenant"}]`. Thoughts?

Reply:
Ouch, I am sorry. I think the change in behaviour is not acceptable because it does change how the migrator sees and runs migrations. I think your original version is the way to go indeed. Can you please revert? And thank you for exploring this path. :)

Reply:
I've reverted [6c36d02](https://github.com/phoenixframework/phoenix_ecto/pull/181/commits/6c36d02e8702a34884cd540b16b01ff12f9a54ab) which means the migrator will run once for the `["a", "b"]` case (although the code to maintain this behaviour is a bit complicated). Oh, did you mean revert back to the original version with the `prefix` option (i.e. to include only the first 2 commits in this PR)?

Reply:
Rollback to the original version. It would require you to list the plug twice but I think that's alright?

Reply:
Reverted to the original version. Yeah, listing the plug twice is fine for this use case (a known dev tenant prefix).